src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,125 @@
 package core.basesyntax;
 
-/**
+import java.util.NoSuchElementException;
+
+/*
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPASITY = 10;
+    private T[] arrayGeneric;
+    private int size;
+
+    public ArrayList() {
+        arrayGeneric = (T[]) new Object[CAPASITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == arrayGeneric.length) {
+            arrayGeneric = resize();
+        }
+        arrayGeneric[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException(
+                    "This index is not suitable for the given array");
+        }
 
+        if (size == arrayGeneric.length) {
+            arrayGeneric = resize();
+        }
+        System.arraycopy(arrayGeneric, index, arrayGeneric, index + 1, size - index);
+        arrayGeneric[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size + list.size() >= arrayGeneric.length) {
+            arrayGeneric = resize();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return arrayGeneric[index];
     }
 
     @Override
     public void set(T value, int index) {
+        checkIndex(index);
+        arrayGeneric[index] = value;
 
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return editArray(index);
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        int index = findRemoveElement(t);
+        if (index != -1) {
+            return editArray(index);
+        }
+        throw new NoSuchElementException("No matching item found");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public T[] resize() {
```

Comment:
```suggestion
    private T[] resize() {
```
please make all "helper" methods private

Reply:
Ok, I will do.